### PR TITLE
[BUG] Admin order create validate vat confirm box

### LINKED ISSUE #29652: [Issue] [BUG] Admin order create validate vat confirm box
This issue is automatically created based on existing pull request: magento/magento2#29466: [BUG] Admin order create validate vat confirm box

---------
Was broken with commit https://github.com/magento/magento2/commit/faa5f9a572ac1ea44750936bd9dc51c20513269d#diff-47459785dc7008bc9145a9609c9c7f6c and never fixed.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Preconditions (*)
1. Magento 2.4-develop

### Steps to reproduce (*)
1. Go to _Stores - Configuration - General - General - Country Options_. Set **Default Country** - _Germany_
2. Expand _Store Information_, select **Country** - Germany. Select any **Region/State**
3. Enter a valid German **VAT Number**. **Validate VAT Number**
4. **Save Config**
5. Go to _Stores - Tax Rules_ click **Add New Tax Rule** 
6. Click **Add New Tax Rate**. Set the following values:
- **Country** - _United Kingdom_
- **Zip/Post is Range** - checkbox _Checked_
- **Range From** - _1_
- **Range To** - _99999999_
- **Rate Percent** - _10_
Click **Save**
7. Specify **Name** _"tax rule"_, for example. Assert that _Rate1_ and previously created rate are marked. **Save Rule**
8. Go to **Customer - Customer Groups** and create a **Customer Group**, with Name _"newgroup"_ for example
9. Go to _Stores - Configuration - Customers - Customer Configuration - _Create New Account Options_ 
10. Set the following values:
- **Enable Automatic Assignment to Customer Group** - _Yes_
- **Default Group** - _Newgroup_
- **Group for Valid VAT ID - Intra-Union** - _Retailer_
- **Validation Error Group** - _General_
- **Validate on Each Transaction** - _Yes_
**Default Value for Disable Automatic Group Changes Based on VAT ID** - _Yes_
**Save Config** clear Magento Cache
11. Go to **Sales - Orders**, click **Create New Order**
12. Click **Create New Customer** 
(Create New Order window appears and selected group is the previously created "newgroup")
13. Change the **Country** to _United Kingdom_
14. Enter a valid  United Kingdom **VAT Number** and click **Validate VAT Number**

### Actual result (*)
Nothing happened after clicking **Validate VAT Number**
before

### Expected result (*)
_... Would you like to change the Customer Group for this order?_ 
vat_validation

And after clicking **OK** the group is changed
vat_validation_retailer

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by Swahjak**
Was broken with commit https://github.com/magento/magento2/commit/faa5f9a572ac1ea44750936bd9dc51c20513269d#diff-47459785dc7008bc9145a9609c9c7f6c and never fixed.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#29652: [BUG] Admin order create validate vat confirm box

### Manual testing scenarios (*)

1. Setup a store using the magento guidelines on VAT configuration: https://docs.magento.com/user-guide/tax/vat-validation-configure.html
2. Setup store configuration that uses a origin address that's within the EU with a valid VAT ID 
3. Start creating an "intra-community/union" order. Meaning; The store address **must** be an EU country and the order shipping address **must** be an EU country which is **not** the origin (store) country
4. Enter at least a valid e-mail address and the billing country
4. Enter a valid VAT ID for on the order
5. Click 'validate'

## Expected result
6. The order creation page should show a prompt asking whether or not the customer group should be changed

## Actual result
6. Nothing happens

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29652: [BUG] Admin order create validate vat confirm box

### CODE REVIEW

File: app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js [hunk 1]
```diff
@@ -1493,14 +1493,20 @@ define([
...
-                    this.accountGroupChange();
-                }
+                confirm({
+                    content: confirmText,
+                    actions: {
+                        confirm: function() {
+                            $$('#' + groupIdHtmlId + ' option').each(function (o) {
+                                o.selected = o.readAttribute('value') == groupId;
+                            });
+                            self.accountGroupChange();
```
Comment (VladimirZaets) [inline app/code/Magento/Sales/view/adminhtml/web/order/create/scripts.js]: Please bind context instead of using `self` variable.

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @Swahjak, thank for collaboration. Please, sign CLA, otherwise, we can't process your pull request and please take a look to my review comment

**Comment by Swahjak** [comment]
@magento run all tests

**Comment by sidolov** [comment]
@magento create issue

**Review by gabrieldagama — CHANGES_REQUESTED** [review]
Hi @Swahjak, thanks for your pull request! It looks good! All our PR must be covered with tests, would you be able to cover this with a test? I believe MFTF tests would be enough to make sure the functionality works.

Let me know if you are available for it. Thanks.

**Comment by gabrieldagama** [comment]
The risk was set to`low` due to: the suggested changes shouldn't impact other areas.

**Comment by engcom-Hotel** [comment]
Hello @Swahjak, I can help with MFTF. Could you please provide steps for reproducing.

**Comment by Swahjak** [comment]
@engcom-Hotel sorry for my absence. I've been busy. I've never worked with MFTF and to be honest I'm currently not in a position to get started on it, so thank you for your help. I have updated the pull requests description to better reflect the steps to reproduce this issue.

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento Functional Tests B2B, Functional Tests CE

**Review by gabrieldagama — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @gabrieldagama, thank you for the review.
[ENGCOM-8287](https://jira.corp.magento.com/browse/ENGCOM-8287) has been created to process this Pull Request
:eight_spoked_asterisk: @gabrieldagama, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@Swahjak thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

1. Go to _Stores - Configuration - General - General - Country Options_. Set **Default Country** - _Germany_
2. Expand _Store Information_, select **Country** - Germany. Select any **Region/State**
3. Enter a valid German **VAT Number**. **Validate VAT Number**
4. **Save Config**
5. Go to _Stores - Tax Rules_ click **Add New Tax Rule** 
6. Click **Add New Tax Rate**. Set the following values:
- **Country** - _United Kingdom_
- **Zip/Post is Range** - checkbox _Checked_
- **Range From** - _1_
- **Range To** - _99999999_
- **Rate Percent** - _10_

5. Go to Cusromer - 


5. Go to _Stores - Configuration - Customers - Customer Configuration - Create New Account Options_ 
6. Set the following values:
- **Enable Automatic Assignment to Customer Group** - _Yes_
- **Default Group** - _Newgroup_
- **Group for Valid VAT ID - Intra-Union** - _Retailer_
- **Validation Error Group** - _General_
- **Validate on Each Transaction** - _Yes_
**Save Config** clear Magento Cache
7.

**Pull request merged**